### ISSUE TITLE
SWBD run.sh, text preprocessing

### ISSUE THREAD

**Issue by Dreyn** [espnet/espnet#726]
https://github.com/espnet/espnet/blob/049b8236b160d2e48cfd13dd2891363ade9a13ec/egs/swbd/asr1/run.sh#L107

Hello,

There is, in this line, a Perl command :
```bash
perl -pe 's| \(\%.*\)||g'
```
If I'm correct, this command is executed on the transcriptions files, in order to take off every parenthesis like **(%HESITATION)**. But, Perl beeing greedy by default, if a transcription line had to groups of parenthesis with text between, the two groups of parenthesis are taken as one big, with the text.

For example, when I launch the script, the following line :
"This is huh (%HESITATION) a not so (%HESITATION) inspired example"
Becomes :
"This is huh inspired example"

I solved this issue by adding a question mark '?' behind the the star in the regular expression :
```bash
perl -pe 's| \(\%.*?\)||g'
```
It makes the research no greedy anymore, so that it will look for the shortest expression instead of the longest.

This isn't the only perl command using a greedy research (there is another one just after), but I don't recall what the others do...

**Comment by kan-bayashi** [comment]
Thank you for your report, @Dreyn!
I checked this issue is reproduced.
We will fix it accordingly.

**Comment by Emrys365** [comment]
Hello,

I encountered a problem related to this issue.
When testing the model trained with default config in SWBD, I got the following results:

| No. | model | (eval2000) WER/S.Err | (eval2000/callhm) WER/S.Err | (eval2000/swbd) WER/S.Err |
| --- | --- | --- | --- | --- |
| 1 | blstmp (default) | 39.8/68.8 | 43.5/73.0 | 35.7/62.6 |
| 2 |   + fix issue 726 | 33.4/68.3 | 42.0/72.7 | 24.6/61.9 |
| 3 |   + no convert stm | 31.5/67.1 | 40.6/72.1 | 22.4/59.9 |

The first model was trained using the original `run.sh`, without fixing this issue.
The second model was trained after fixing this issue in stage 0.
The third model was trained in the same way as the first one, but without replacing `stm` file in the decoding stage.

> In the decoding stage, `local/score_sclite.sh` replaces the original `stm` file with a new one, where the parentheses and `(%HESITATION)` are removed. The related code is:
> 
```bash
score_dir=${dir}/scoring
ctm=${score_dir}/hyp.ctm
stm=${score_dir}/ref.stm        # replaced by the original stm in model 3
mkdir -p ${score_dir}
if [ ${stage} -le 0 ]; then
    if [ -z ${dict} ]; then
        # Assuming trn files exist
        ref=${dir}/ref.wrd.trn
        hyp=${dir}/hyp.wrd.trn
        trn2stm.py --orig-stm ${data}/stm ${ref} ${stm}   # commented in model 3
        trn2ctm.py ${hyp} ${ctm}
    else
        ref=${dir}/ref.trn
        hyp=${dir}/hyp.trn
        json2sctm.py ${dir}/data.json ${dict} --orig-stm ${data}/stm --stm ${stm} --refs ${ref} --ctm ${ctm} --hyps ${hyp}
    fi
fi
```
>

We can see the results are greatly improved after fixing the issue. But model 3 is even better than model 2. Does that indicate it is not necessary to do the removal of parentheses?

**Comment by sw005320** [comment]
Thanks, @Emrys365.
This issue is only related to the test set, and during training, No 2. and 3 should not have so much difference except for the validation loss/accuracies. During recognition, we should use the correct reference (#726) and even if the score is better, we should use No.2.

Maybe I misunderstood No.3. Could you let me know how No.2 and No.3 are different during recognition/scoring given we use the same model?

**Comment by Emrys365** [comment]
> Thanks, @Emrys365.
> This issue is only related to the test set, and during training, No 2. and 3 should not have so much difference except for the validation loss/accuracies. During recognition, we should use the correct reference (#726) and even if the score is better, we should use No.2.
> 
> Maybe I misunderstood No.3. Could you let me know how No.2 and No.3 are different during recognition/scoring given we use the same model?

Sorry I didn't make it clear.
No.2 and No.3 are different only in the decoding stage. In No.3, I modified `local/score_sclite.sh` to keep the original `stm` as the reference, rather than using the following code to remove the parentheses in it:

```bash
trn2stm.py --orig-stm ${data}/stm ${ref} ${stm}
```

And the rest parts are the same for both No.2 and No.3.

**Comment by sw005320** [comment]
I see. Is this (keeping the parentheses) a normal way? Could you check the Kaldi recipe to double check it?
